test/README.md

```diff
@@ -0,0 +1,49 @@
+# Functional Tests
+
+This directory holds the functional tests for PBnJ.
+
+## Usage
+
+First populate a `resources.yaml` with BMC resource information. This example is also found in `resources.example.yaml`. This file needs to exist in the top level of the `test` directory.
+```yaml
+---
+resources:
+- ip: 10.10.10.10 # BMC IP
+  username: admin # BMC username
+  password: admin # BMC password
+  vendor: HP # BMC vendor
+  useCases: # use cases for each RPC test
+    power: # should correspond to an RPC name
+    - happyTests # should correspond to the name of a structure that holds test inputs and expected outputs; see table driven tests pattern
+      sadTests # as many of this as needed can be added
+    device:
+    - happyTests
+      sadTests
+- ip: 10.5.5.5
+  username: ADMIN
+  password: ADMIN
+  vendor: supermicrox10
+  useCases: 
+    power: 
+    - happyTests
+      sadTests
+    device:
+    - happyTests
+      sadTests
+```
+
+Run the tests
+```bash
+make test-functional
+# under the hood it calls: go test -v ./test/... --tags=functional -config 'resources.yaml'
+```
+
+## Development
+
+* The functional tests use the standard `Go` testing library.
+* Test files should follow normal `go test` patterns and be named with `_test.go`
+* Test files should contain a [build tag](https://golang.org/cmd/go/#hdr-Build_constraints) of `// +build functional` at the start of the file.
```

Comment:
why?

Reply:
so that `go test ./...` doesn't run the functional tests

Reply:
:facepalm: I thought we were also faking the BMC accesses so thought we could do functional tests all the time. I think maybe its fair to be explicit and call that out :D.

Reply:
faking? no, this is for against real hardware